src/main/java/net/dv8tion/jda/core/utils/PermissionUtil.java

```diff
@@ -420,79 +481,165 @@ public static long getEffectivePermission(Channel channel, Role role)
+        if (override != null)
+            permission = apply(permission, override.getAllowedRaw(), override.getDeniedRaw());
+
         return permission;
     }
 
+    /**
+     * Retrieves the implicit permissions of the specified {@link net.dv8tion.jda.core.entities.Role Role}
+     * in its hosting {@link net.dv8tion.jda.core.entities.Guild Guild} and specific {@link net.dv8tion.jda.core.entities.Channel Channel}.
+     * <br><b>Allowed permissions override denied permissions of {@link net.dv8tion.jda.core.entities.PermissionOverride PermissionOverrides}!</b>
+     *
+     * <p>All permissions returned are explicitly granted to this Role.
+     * <br>Permissions like {@link net.dv8tion.jda.core.Permission#ADMINISTRATOR Permission.ADMINISTRATOR} do not
+     * grant other permissions in this value.
+     * <p>This factor in existing {@link net.dv8tion.jda.core.entities.PermissionOverride PermissionOverrides} if possible.
+     *
+     * @param  channel
+     *         The target channel of which to check {@link net.dv8tion.jda.core.entities.PermissionOverride PermissionOverrides}
+     * @param  role
+     *         The non-null {@link net.dv8tion.jda.core.entities.Role Role} for which to get implicit permissions
+     *
+     * @throws IllegalArgumentException
+     *         If any of the arguments is {@code null}
+     *         or the specified entities are not from the same {@link net.dv8tion.jda.core.entities.Guild Guild}
+     *
+     * @return Primitive (unsigned) long value with the implicit permissions of the specified role in the specified channel
+     *
+     * @since  3.1
+     */
+    public static long getImplicitPermission(Channel channel, Role role)
+    {
+        Args.notNull(channel, "Channel");
+        Args.notNull(role, "Role");
+        checkGuild(channel.getGuild(), role.getGuild(), "Role");
+
+        long permission = 0;
+        PermissionOverride override = channel.getPermissionOverride(role.getGuild().getPublicRole());
+        if (override != null)
+            permission = apply(permission, override.getAllowedRaw(), override.getDeniedRaw());
+        if (role.equals(role.getGuild().getPublicRole()))
+            return permission;
+
+        override = channel.getPermissionOverride(role);
+
+        return override == null
+            ? permission
+            : apply(permission, override.getAllowedRaw(), override.getDeniedRaw());
+    }
+
+    /*
+     * Check whether the specified permission is applied in the bits
+     */
+    private static boolean isApplied(long permissions, Permission permission)
+    {
+        return (permissions & permission.getRawValue()) > 0;
+    }
+
     private static long apply(long permission, long allow, long deny)
     {
-        permission = permission | allow;    //Allow all the things that the cascade of roles allowed
-        permission = permission & (~deny);  //Deny everything that the cascade of roles denied.
+        deny &= ~allow;
```

Comment:
This makes sure that no bits in `deny` are also 1 in `allow`
Example:
`1010 & ~1000` is `1010 & 0111` -> `0010`
This is done because permissions that are allowed override the ones that are denied.